erpnext/stock/doctype/item/item.py

```diff
@@ -486,9 +486,22 @@ def add_default_uom_in_conversion_factor_table(self):
 		[self.remove(d) for d in to_remove]
 
 	def update_show_in_website(self):
+		doc = self.get_doc_before_save()
+
 		if self.disabled:
 			self.show_in_website = False
 
+		elif not self.disabled and self.has_variants and not self.show_in_website and not self.show_in_website == doc.show_in_website:
+			variants = frappe.get_all("Item", filters={"variant_of": self.name}, fields=["item_code"])
+
+			if not variants:
+				return
+
+			if len(variants) <= 30:
```

Comment:
Just enqueue it always, there's no visible feedback either way. We don't want to keep adjusting this value if required.